cluster-autoscaler/metrics/metrics.go

```diff
@@ -426,3 +437,9 @@ func UpdateScaleDownInCooldown(inCooldown bool) {
 		scaleDownInCooldown.Set(0.0)
 	}
 }
+
+// RegisterOldUnregisteredNodesRemoved records number of old unregistered
+// nodes that have been removed by the cluster autoscaler
+func RegisterOldUnregisteredNodesRemoved(nodesCount int, nodegroupId string) {
+	oldUnregisteredNodesRemovedCount.WithLabelValues(nodegroupId).Add(float64(nodesCount))
```

Comment:
Please don't add labels that have potentially high cardinality (many times new node pools are created by scripts with random name). It may cause problems in many metric storage systems. A quote from prometheus doc:

```
CAUTION: Remember that every unique combination of key-value label pairs represents a new time series, which can dramatically increase the amount of data stored. Do not use labels to store dimensions with high cardinality (many different label values), such as user IDs, email addresses, or other unbounded sets of values.
```

Reply:
that's fair, i can imagine the cardinality getting out of control in those situations. it is really helpful from a debugging perspective, but i suppose the metric could lead some to inspect the autoscaler logs or events. i'll remove the label.

Reply:
I think have many small nodepools is not standard behavior.  One compromise is to make this metric optional and keep it off by default.

For the vast majority of clusters, this metric would be highly useful.

Reply:
an interesting thought about adding a flag to enable/disable this metric. i'd be ok with that, my impression is that we have been trying to limit adding more cli flags though. curious to hear what @mwielgus thinks.

Reply:
We have other ways of providing detailed statistics in CA - there is a status configmap and logs. I would rather not copy their functionality in metrics. Or at least I would not add group name labels in just single place - cluster size, etc could also be split across node groups. If you feel strongly about labels, please merge this PR without labels and start a new PR/discussion about consistent metrics labeling.

Reply:
thanks Marcin, appreciate the discussion. i'm going to repost without the label because i think this is useful as a metric source regardless of that detail.